src/net/tls.cc

```diff
@@ -887,8 +896,9 @@ class tls::reloadable_credentials_base {
             }
         }
         void do_callback(std::exception_ptr ep = {}) {
-            if (_cb && !_files.empty()) {
-                _cb(boost::copy_range<std::unordered_set<sstring>>(_files | boost::adaptors::map_keys), std::move(ep));
+            if (_cb && !_files.empty() && _creds) {
+                const auto &creds = _creds->as_certificate_credentials();
+                _cb(boost::copy_range<std::unordered_set<sstring>>(_files | boost::adaptors::map_keys), creds, std::move(ep));
```

Comment:
How is the set of files relevant for the receiver of the call? Given that we provide no relation to the actual files used by the resulting credentials, it seems a receiver must know them anyway to match them to anything, no?
Here it is not even changed files, but all watched. Seems like unneeded overhead...

Reply:
I hadn't given it much thought to be honest. Intention was for this to be a non-breaking change. Is your suggestion to remove this now that the credentials are included with the callback?

Reply:
Good stuff, thanks :)